### PR TITLE
Created a function to check if the inputed event is overlapping the previous events

### PR DESCRIPTION
**Pull request by YvonneAriri**
Description:

-Created a function that checks if the user's input overlaps the preceding events.
-Integrated it to the event form so it doesn't allow the user to create an event if the event is overlapping with the preceding events.

Milestone: Implemented the code to do a check if event to be created is overlapping with any other event of the user before it is created.

Test plan:
Eveprep12

### CODE REVIEW

File: event-app-api/merge_intervals/merge_intervals.js [hunk 1]
```diff
@@ -40,3 +38,19 @@ export function findGaps(mergedIntervals) {
...
+
+// Function to check if an interval overlaps the other and vice versa
+export function isOverlapping(inputedEvent, events) {
+  for (let i = 0; i < events.length; i++) {
+    if (inputedEvent.starttime <= events[i].starttime) {
+      if (inputedEvent.endtime > events[i].starttime) {
+        return true;
+      }
+    } else {
+      if (inputedEvent.endtime <= events[i].starttime) {
```
Comment (gyngve) [inline event-app-api/merge_intervals/merge_intervals.js:56]: this check is not necessary, assuming yout intervals are well-defined (starttime < endtime)

Comment (gyngve) [reply event-app-api/merge_intervals/merge_intervals.js:56]: it is also incorrect: with a unit test, you should be able to see that it doesn't work as intended

Comment (gyngve) [reply event-app-api/merge_intervals/merge_intervals.js:56]: no need to go by index, use iteration:

for(event in events) {
  if (inputedEvent.starttime <= event.starttime) ...

**Review by gyngve — CHANGES_REQUESTED** [review]

File: event-app-api/merge_intervals/merge_intervals.js (hunk 1 above)
Comment (sophia129) [reply event-app-api/merge_intervals/merge_intervals.js:56]: This logic can be simplified. There are only 2 conditions when two events are overlapping: Event 1 starts between Event 2 start and end time, OR Event 1 ends between Event 2 start and end time

**Pull request merged**